src/pq.py

```diff
@@ -0,0 +1,68 @@
+"""Implement a priority queue."""
+import sys
+
+if sys.version_info[0] == 3:
```

Comment:
Also, what happens here when python4 comes out?  You can try/except stuff like this also (if you need to, which you don't here)